### ISSUE TITLE
Errors compiling with gcc-10

### ISSUE THREAD

**Issue by waebbl** [materialx/MaterialX#423]
On Gentoo, I'm having issues, building MaterialX-1.37.0 using gcc-10.1.0

```
[3/196] /usr/bin/x86_64-pc-linux-gnu-g++ -DMATERIALX_BUILD_VERSION=0 -DMATERIALX_MAJOR_VERSION=1 -DMATERIALX_MINOR_VERSION=37 -DMATERIALX_OSLC_EXECUTABLE=\"\" -DMATERIALX_OSL_INCLUDE_PATH=\"\" -DMATERIALX_TESTRENDER_EXECUTABLE=\"\" -I/var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/..  -O2 -pipe -march=bdver2 -fstack-protector-strong -fstack-check -fPIC   -Wall -Wno-missing-braces -std=gnu++11 -MD -MT source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o -MF source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o.d -o source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o -c /var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/Element.cpp
FAILED: source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DMATERIALX_BUILD_VERSION=0 -DMATERIALX_MAJOR_VERSION=1 -DMATERIALX_MINOR_VERSION=37 -DMATERIALX_OSLC_EXECUTABLE=\"\" -DMATERIALX_OSL_INCLUDE_PATH=\"\" -DMATERIALX_TESTRENDER_EXECUTABLE=\"\" -I/var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/..  -O2 -pipe -march=bdver2 -fstack-protector-strong -fstack-check -fPIC   -Wall -Wno-missing-braces -std=gnu++11 -MD -MT source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o -MF source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o.d -o source/MaterialXCore/CMakeFiles/MaterialXCore.dir/Element.cpp.o -c /var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/Element.cpp
/var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/Element.cpp: In member function ‘virtual std::pair<int, int> MaterialX::Element::getVersionIntegers() const’:
/var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/Element.cpp:154:17: error: ‘invalid_argument’ in namespace ‘std’ does not name a type
  154 |     catch (std::invalid_argument&)
      |                 ^~~~~~~~~~~~~~~~
/var/tmp/portage/media-libs/MaterialX-1.37.0/work/MaterialX-1.37.0/source/MaterialXCore/Element.cpp:157:17: error: ‘out_of_range’ in namespace ‘std’ does not name a type
  157 |     catch (std::out_of_range&)
      |                 ^~~~~~~~~~~~
```

When using gcc < 10, the build succeeds, so I thought it might be related to the change in `-fcommon` defaulting to `-fno-common` in gcc-10, but simply adding the `-fcommon` flag when building didn't solve the issue.

**Comment by jstone-lucasfilm** [comment]
Good catch, and I believe this is caused by the removal of the `stdexcept` header from standard libraries such as `string` in GCC 10:

https://gcc.gnu.org/gcc-10/porting_to.html

Assuming that is the issue, then the fix would be to add the following line to the block of standard header includes in Element.cpp:

`#include <stdexcept>`

Would you be interested in trying this fix in your build environment, and posting a pull request to MaterialX GitHub if it succeeds?  The one caveat is that you'd need to sign a Contributor License Agreement as described here:

https://github.com/materialx/MaterialX/blob/master/CONTRIBUTING.md

**Comment by waebbl** [comment]
Yes, I can test this.

**Comment by jstone-lucasfilm** [comment]
Thanks for bringing this to our attention, @waebbl!  The fix has been implemented in #433 and merged to master.

**Comment by waebbl** [comment]
Thanks for fixing this @jstone-lucasfilm. The only case I found, was the one in `src/MaterialXCore/Element.cpp`, but then got side-tracked by other projects.
